# HG changeset patch
# User Kai T. Ohlhus <k.ohlhus@gmail.com>
# Date 1490256695 -3600
#      Thu Mar 23 09:11:35 2017 +0100
# Branch stable
# Node ID 71a749eeb7d3878411e791cfacba8791886c6dae
# Parent  336f2e8331fa77caa2a89ebb794b32fc6e08d939
test: publish to a temporary directory to ensure write access.

* test/publish/publish.tst: Don't rely on the test directory to be installed in
a writeable location.  To ensure the integration test to be successful, publish
to a temporary directory.

diff --git a/test/publish/publish.tst b/test/publish/publish.tst
--- a/test/publish/publish.tst
+++ b/test/publish/publish.tst
@@ -1,8 +1,26 @@
+## Copyright (C) 2016-2017 Kai T. Ohlhus
+##
+## This file is part of Octave.
+##
+## Octave is free software; you can redistribute it and/or modify it
+## under the terms of the GNU General Public License as published by
+## the Free Software Foundation; either version 3 of the License, or
+## (at your option) any later version.
+##
+## Octave is distributed in the hope that it will be useful, but
+## WITHOUT ANY WARRANTY; without even the implied warranty of
+## MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+## GNU General Public License for more details.
+##
+## You should have received a copy of the GNU General Public License
+## along with Octave; see the file COPYING.  If not, see
+## <http://www.gnu.org/licenses/>.
+
 ## test for publish
 
 %!test
 %! visibility = get (0, "defaultfigurevisible");
 %! toolkit = graphics_toolkit ();
 %! unwind_protect
 %!   if (ispc () || ! __have_feature__ ("OSMESA"))
 %!     try
@@ -11,21 +29,24 @@
 %!       ## The system doesn't support OSMESA or gnuplot for drawing hidden
 %!       ## figures.  Just return and have test marked as passing.
 %!       return;
 %!     end_try_catch
 %!   endif
 %!   set (0, "defaultfigurevisible", "off");
 %!
 %!   scripts = dir ("test_script*.m");
+%!   tmpDir = tempname ();
+%!   mkdir (tmpDir);
+%!   opts.outputDir = tmpDir;
 %!   for fname = {scripts.name}
-%!     publish (fname{1});
+%!     publish (fname{1}, opts);
 %!   endfor
 %!   confirm_recursive_rmdir (false, "local");
-%!   rmdir ("html", "s");
+%!   rmdir (tmpDir, "s");
 %! unwind_protect_cleanup
 %!   set (0, "defaultfigurevisible", visibility);
 %!   graphics_toolkit (toolkit);
 %! end_unwind_protect
 
 ## test for grabcode
 
 %!test
@@ -37,22 +58,26 @@
 %!       graphics_toolkit ("gnuplot");
 %!     catch
 %!       ## The system doesn't support OSMESA or gnuplot for drawing hidden
 %!       ## figures.  Just return and have test marked as passing.
 %!       return;
 %!     end_try_catch
 %!   endif
 %!   set (0, "defaultfigurevisible", "off");
-%!
-%!   publish ("test_script.m");
+%!   ## Create temporary directory
+%!   tmpDir = tempname ();
+%!   mkdir (tmpDir);
+%!   opts.outputDir = tmpDir;
+%!   ## Call publish and grabcode
+%!   publish ("test_script.m", opts);
 %!   str1 = fileread ("test_script.m");
-%!   str2 = grabcode ("html/test_script.html");
+%!   str2 = grabcode (fullfile (tmpDir, "test_script.html"));
 %!   confirm_recursive_rmdir (false, "local");
-%!   rmdir ("html", "s");
+%!   rmdir (tmpDir, "s");
 %!   ## Canonicalize strings
 %!   str1 = strjoin (deblank (strsplit (str1, "\n")), "\n");
 %!   str2 = strjoin (deblank (strsplit (str2, "\n")), "\n");
 %!   assert (hash ("md5", str1), hash ("md5", str2));
 %! unwind_protect_cleanup
 %!   set (0, "defaultfigurevisible", visibility);
 %!   graphics_toolkit (toolkit);
 %! end_unwind_protect
